pexec/managed_process.go

```diff
@@ -265,6 +268,14 @@ func (p *managedProcess) manage(stdOut, stdErr io.ReadCloser) {
 	default:
 	}
 
+	// Run onCrashHandler if it exists. Do not attempt restart if onCrashHandler
+	// returns false.
+	if p.onCrashHandler != nil {
+		if shouldRestart := p.onCrashHandler(); !shouldRestart {
+			return
+		}
+	}
```

Comment:
```suggestion
	if p.onCrashHandler != nil && !p.onCrashHandler() {
		return
	}
```
Pretty sure that simplifies it correctly.